internal/wasm/sdk/test/e2e/external_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/internal/wasm/sdk/opa"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/test/cases"
+	"github.com/open-policy-agent/opa/types"
+	"github.com/open-policy-agent/opa/util"
+)
+
+var caseDir = flag.String("case-dir", "../../../../../test/cases/testdata", "set directory to load test cases from")
+var exceptionsFile = flag.String("exceptions", "./exceptions.yaml", "set file to load a list of test names to exclude")
+
+var exceptions map[string]string
+
+func TestMain(m *testing.M) {
+	exceptions = map[string]string{}
+
+	bs, err := ioutil.ReadFile(*exceptionsFile)
+	if err != nil {
+		fmt.Println("Unable to load exceptions file: " + err.Error())
+		os.Exit(1)
+	}
+	err = util.Unmarshal(bs, &exceptions)
+	if err != nil {
+		fmt.Println("Unable to parse exceptions file: " + err.Error())
+		os.Exit(1)
+	}
+
+	addTestSleepBuiltin()
+
+	os.Exit(m.Run())
+}
+
+func TestWasmE2E(t *testing.T) {
+
+	ctx := context.Background()
+
+	for _, tc := range cases.MustLoad(*caseDir).Sorted().Cases {
+		if tc.Note != "withkeyword/invalidate comprehension cache" {
+			continue
+		}
+		t.Run(tc.Note, func(t *testing.T) {
+			t.Log("Test File: " + tc.Filename)
```

Comment:
nit: can we use the filename here if the # of cases in the file == 1? This will be much more readable IMO. If the file has multiple cases it could be `<filename>-<note>` or something.

Reply:
Sure, if you prefer that format that is fine with me.

Reply:
To avoid having to scan over the tests to find if there is >1 in each (all the generated ones we had originally are 1 per file, newer hand written ones can/do contain more) I opted for `<filename>/<note>` which yields test names like:

```
=== RUN   TestWasmE2E/test/cases/testdata/aggregates/test-aggregates-0004.yaml/aggregates/count_keys_virtual
```

Kind of verbose, but gives a lot of information.

Reply:
The verbosity is fine IMO. I forgot that we flatten the cases in that package. Looks good!